client/base_client.go

```diff
@@ -11,4 +11,6 @@ type BaseClient interface {
 	SetTimeout(timeout time.Duration)
 	SendRequest(method string, rawURL string, data url.Values,
 		headers map[string]interface{}, body ...byte) (*http.Response, error)
+	SetOauth(auth OAuth)
+	GetOAuth() OAuth
```

Comment:
Typically getters in Go just have the name of the thing being retrieved 

```suggestion
	OAuth() OAuth
```